File: ReadyTasks/FSharp/ProjectEuler/Task_27/Task_27/Program.fs
open System

let isPrime num =
    if (num < 2L) then
        false
    else
        Seq.initInfinite (fun x -> (int64)(x + 2)) |> 
        Seq.takeWhile (fun x -> x * x <= num) |>
        Seq.forall (fun x -> num % x > 0L)

let countMaxPrimes a b =
    Seq.initInfinite (fun x -> x) |>
    Seq.takeWhile (fun x -> x * x + a * x + b |> int64 |> isPrime) |>
    Seq.length

let getResult maxA maxB =
    [-maxA..maxA] |> List.allPairs [-maxB..maxB] |>
    List.maxBy (fun (a,b) -> countMaxPrimes a b) |>
    (fun (a,b) -> a * b)

[<EntryPoint>]
let main argv =
    printfn "%A" (getResult 999 1000)
    0 // return an integer exit code

File: ReadyTasks/CSharp/ProjectEuler/Task_27/Task_27/Program.cs
using System;

namespace Task_27
{
    class Program
    {
        static bool IsPrime(int numb)
        {
            for (int i = 2; i * i <= numb; i++)
            {
                if (numb % i == 0)
                {
                    return false;
                }
            }
            return numb > 1;
        }

        static int GetMaxPrimes(int a, int b)
        {
            int n = 0;
            while (IsPrime(n * n + a * n + b))
            {
                n++;
            }
            return n;
        }

        static int GetResult(int minA, int maxA, int minB, int maxB)
        {
            int res = 0;
            int a = 0;
            int b = 0;
            for (int i = minA; i <= maxA; i++)
            {
                for (int j = minB; j <= maxB; j++)
                {
                    int newRes = GetMaxPrimes(i, j);
                    if (newRes > res)
                    {
                        res = newRes;
                        a = i;
                        b = j;
                    }
                }
            }
            return a * b;
        }

        static void Main(string[] args)
        {
            Console.WriteLine(GetResult(-1000,1000,-1000,1000));
        }
    }
}
